### DISCUSSION TITLE
Setting up kubernetes on Azure for Nebari use

### DISCUSSION THREAD

**Discussion by rsignell-usgs** [nebari-dev/nebari#1846] [Q&A] [answered]
I'm working with folks at NATO to see if we can get Nebari deployed in their restricted Azure environment.   They currently have the ability to set up a kubernetes cluster, which seemed like perhaps the best way to proceed, following something like [the instructions for AWS to deploy on existing infrastructure](https://www.nebari.dev/docs/how-tos/nebari-kubernetes/)?  

I asked them to set up a Kubernetes cluster and they responded with these questions:
 
AKS cluster specs:

* How many node pools do you want to have ?
* How many max nodes should each node pool have ?
* What will be the max number of pods per node ?
* What will be the min number of nodes per pool ?
* Do you want any specific k8s version?
* Do you want cluster autoscaler to be enabled ?
* Do you want any specific Azure AD group to be assigned as Contributors ?
* What should be the node size ?

Are there some good starting values that could get us going?

**Comment by iameskild** [comment]
Hi @rsignell-usgs, following that guide for existing infrastructure is a good place to start. Here's a few guidelines for the questions they asked:

- How many node pools do you want to have ?
  - 3 node pools
    - `general` - "Standard_D8_v3"
    - `worker` - "Standard_D4_v3"
    - `user` - "Standard_D4_v3"
- How many max nodes should each node pool have ?
  - `general` - 1 max node only
  - `worker` - 5 nodes
  - `user` - 5 nodes
- What will be the max number of pods per node ?
  - Using the default profiles for Jupyter, you can fit two `Small instance` or one `Medium Instance` (pods) on one of the `user` nodes above.
  - Using the default profiles for Dask-Workers, you can fit two `Small Worker` or one `Medium Worker` (pods) on one of the `worker` node above
- What will be the min number of nodes per pool ?
  - The `general` node group requires 1 and only 1 node running at all times
  - You can try setting the `worker` and `user` min nodes to 0
- Do you want any specific k8s version?
  - Kubernets version 1.24 or under (we need to upgrade to a later version soon)
- Do you want cluster autoscaler to be enabled ?
  - Yes   
- Do you want any specific Azure AD group to be assigned as Contributors ?
  - I'm not sure what this means exactly but the person deploying will likely need at Owner privileges to deploy the cluster
- What should be the node size ?
  - See first answer.

For reference, this is what the [default values for Azure](https://github.com/nebari-dev/nebari/blob/596083a640585bddedd5641a5c630309becf393a/src/_nebari/initialize.py#L239) deploy look like when the `nebar-config.yaml ` is first initialized with Azure.

I hope that helps, let me know if they have any other questions :)

**Reply by rsignell-usgs** [reply, marked as answer]
@iameskild thanks so much!  That should do it !  I'll post back here if we have success!
